sympy/ntheory/primetest.py

```diff
@@ -110,6 +110,24 @@ def mr(n, bases):
     return True
 
 
+def isprimeE(n, b):
+    """isprimeE(n ,b) - Test whether n is prime or an Euler pseudoprime to base b."""
+	if (not mr(n, b)): return False
+    
+	r = n-1
+	while (r % 2 == 0): r //= 2
```

Comment:
Make sure you are working with ints and then this can be `r >>= trailing(r)`

Reply:
Yeah, I have changed it now,
thanks for the review!!